addons/wowl/static/src/views/legacy/legacy_view_dialogs.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, tags, hooks } from "@odoo/owl";
+import { OdooEnv, Type, ViewProps } from "../../types";
+import { Dialog } from "../../components/dialog/dialog";
+const { useRef } = hooks;
+
+const SIZE_CLASSES: { [key: string]: string } = {
+  "extra-large": "modal-xl",
+  large: "modal-lg",
+  small: "modal-sm",
+};
+
+interface LegacyViewDialogProps {
+  ComponentProps: ViewProps;
+  Component: Type<Component<{}, OdooEnv>>;
+}
+
+export class LegacyViewDialog extends Component<LegacyViewDialogProps, OdooEnv> {
+  static template = tags.xml`
+    <Dialog t-props="dialogProps" t-ref="dialogRef">
+      <t t-component="props.Component" t-props="props.ComponentProps" t-ref="compRef"/>
+      <t t-set-slot="buttons"><!-- no buttons for now --></t>
+    </Dialog>
+  `;
+  static components = { Dialog };
+  compRef = useRef("compRef") as any;
+  dialogRef = useRef("dialogRef") as any;
+  dialogProps: { title?: string; size?: string } = {};
+
+  constructor() {
+    super(...arguments);
+    const { ComponentProps } = this.props;
+    const size = SIZE_CLASSES[ComponentProps?.action?.context.dialog_size || ""];
+    if (size) {
+      this.dialogProps.size = size;
+    }
+    const title = ComponentProps?.action?.name;
+    if (title) {
+      this.dialogProps.title = this.env._t(title);
+    }
+  }
+
+  mounted() {
+    this.updateDialog();
+  }
+
+  patched() {
+    this.updateDialog();
+  }
+
+  updateDialog() {
+    const modal = this.dialogRef.comp.modalRef.el as HTMLElement;
+
+    const dialogFooter = modal.querySelector("footer") as HTMLElement;
+    this.compRef.comp.compRef.comp.widget.renderButtons(dialogFooter);
+
```

Comment:
const compFromUpdateUI = this.compRef.comp
    const compFromViewAdapter = compFromUpdateUI.comp